src/components/adaptive-sidebar/adaptive-sidebar.style.ts

```diff
@@ -0,0 +1,70 @@
+import styled, { css } from "styled-components";
+
+import { margin, MarginProps, padding, PaddingProps } from "styled-system";
+
+import Box from "../box";
+import Sidebar, { SidebarProps } from "../sidebar";
+
+import { AdaptiveSidebarProps } from "./adaptive-sidebar.component";
+
+const getColors = (
+  backgroundColor: AdaptiveSidebarProps["backgroundColor"],
+) => {
+  switch (backgroundColor) {
+    case "app":
+      return `
+        background-color: var(--colorsUtilityMajor025);
+        color: var(--colorsUtilityYin090);
+      `;
+    case "black":
+      return `
+        background-color: var(--colorsUtilityYin100);
+        color: var(--colorsUtilityYang100);
+      `;
+    case "white":
+    default:
+      return `
+        background-color: var(--colorsUtilityYang100);
+        color: var(--colorsUtilityYin090);
+      `;
+  }
+};
+
+type StyledAdaptiveSidebarProps = Pick<
+  AdaptiveSidebarProps,
+  "backgroundColor" | "borderColor" | "height" | "width"
+> &
+  MarginProps &
+  PaddingProps;
+
+const StyledAdaptiveSidebar = styled(Box)<StyledAdaptiveSidebarProps>`
+  ${({ backgroundColor, borderColor, height, width }) => css`
+    ${getColors(backgroundColor)}
+    ${borderColor &&
+    borderColor !== "none" &&
```

Comment:
question: are we supporting any colour here or are should we be limiting it to token values? If it's tokens only we could go with:

```
${borderColor && `border-left: 1px solid var(${borderColor});`}
```

The above assumes that if `none` is passed we don't pass the borderColor down so it would be undefined, it's more than fine to keep the check you have though